awx/ui_next/src/components/Schedule/shared/buildRuleObj.js

```diff
@@ -0,0 +1,101 @@
+import { t } from '@lingui/macro';
+import { RRule } from 'rrule';
+import { getRRuleDayConstants } from '@util/dates';
+
+export default function buildRuleObj(values, i18n) {
```

Comment:
You could wrap this function in `withI18n` and remove the HOC from ScheduleAdd and ScheduleEdit.